src/amchart/Pie.js

```diff
@@ -93,7 +92,7 @@
         
         this.pieAlpha().forEach(function(d,i) {
             if (typeof(this._chart.chartData[i])==='undefined') { 
-                this._chart.chartData[i] = {} 
+                this._chart.chartData[i] = {};
```

Comment:
In this specific case you need it at the end of every assignment (once this PR is pulled the answer will be "whenever JSHint tells you"<g>).

I think the JS rule is that it is needed to separate JS statements, which is why it is not needed  at the end of code blocks (function xxx() {}, if {} and switch {}) or for the the last statement within a code block.  FWIW JSHint will insist you have them for the last statement in a code block (which is good with me!).